modules/javafx.graphics/src/main/java/com/sun/javafx/font/PrismFontFile.java

```diff
@@ -1374,4 +1399,112 @@ public boolean equals(Object obj) {
     public int hashCode() {
         return filename.hashCode() + (71 * fullName.hashCode());
     }
+
+
+    private boolean checkedColorTables;
+    private boolean hasColorTables;
+    private synchronized boolean fontSupportsColorGlyphs() {
+       if (checkedColorTables) {
+           return hasColorTables;
+       }
+       hasColorTables =
+           getDirectoryEntry(sbixTag) != null ||
+           getDirectoryEntry(colrTag) != null;
+       checkedColorTables = true;
+
+       return hasColorTables;
+    }
+
+    public boolean isColorGlyph(int glyphID) {
+        if (!fontSupportsColorGlyphs()) {
+            return false;
+        }
+        if (getDirectoryEntry(sbixTag) != null) {
+            return isSbixGlyph(glyphID);
+        }
+        return false;
+   }
+
+
+   private static int USHORT_MASK = 0xffff;
+   private static int UINT_MASK   = 0xffffffff;
+
+   static class ColorGlyphStrike {
+
+       private int ppem;
+       private int ppi;
+       private int dataOffsets[];
+
+       ColorGlyphStrike(int ppem, int ppi, int[] offsets) {
+           this.ppem = ppem;
+           this.ppi  = ppi ;
+           dataOffsets = offsets;
+       }
+
+       boolean hasGlyph(int gid) {
+           if (gid >= dataOffsets.length) {
+              return false;
+           }
+           return dataOffsets[gid] < dataOffsets[gid+1];
+       }
+   }
+
+   ColorGlyphStrike[] sbixStrikes = null;
+
+   private boolean isSbixGlyph(int glyphID) {
+       if (sbixStrikes == null) {
+           synchronized (this) {
```

Comment:
This looks like a double-checked locking pattern, which is not guaranteed to be thread-safe.

Suggestion: move the `synchronized` before the `if`.

Reply:
or check for null again after synchronized(this)